internal/handlers/iso.go

```diff
@@ -44,7 +44,7 @@ func (h *isoHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	ignition, lastModified, statusCode, err := h.client.ignitionContent(r, imageID)
+	ignition, lastModified, statusCode, err := h.client.ignitionContent(r, imageID, params.imageType)
```

Comment:
Won't the older versions of `assisted-service` reject this request?
If I'm right, we should probably catch this response and make sure we log something meaningful for users about these circumstances.

Reply:
Something along the lines of "The version of assisted service in use does not permit override of discovery_iso_type, if you wish to generate the ISO based on a specific type, you will need to edit the infraenv" ?

Reply:
It's perfectly valid to make a request such as 

`curl   --silent   --header "Authorization: Bearer $(ocm token)"   "https://api.openshift.com/api/assisted-install/v2/infra-envs/9f49dd92-80b7-4995-84f5-2fac5b535cb3/downloads/files?file_name=discovery.ign&somerandomparam=1" | jq`

In the example above, somerandomparam will be ignored unless it is supported by assisted-service.

This demonstrates that the change is backwards compatible I think?